### DISCUSSION TITLE
Next.js Redirects behave different when deployed on Vercel?

### DISCUSSION THREAD

**Discussion by ScottGrun** [vercel/next.js#35836] [Help]
Does anyone know if there is any difference in the way Vercel runs 'next build' vs running it locally? I am trying to redirect from some dead pages in my next.config.js working and I am redirecting from a source of one catch all route to a destination of another catch all route. It works locally but when deployed Vercel it doesn't ? I have tried redirects and rewrites in the ` next.config.js ` and `vercel.json` . Furthermore it only seems not work for the catch all routes any other redirects work fine when deployed ? For the time being we have got this to work using middleware but that seems heavy handed for handling redirects?

I have also compiled the code locally with `next build` and `npm run start` and it still works locally but not on Vercel? I also confirmed t that all .env vars are the same.

## Expected Behavior:
Catch all routes that have redirects or rewrites do not redirect and instead 404 when other routes redirect fine.

## Current Behavior:
Catch all routes that have redirects or rewrites configured in `next.config.js` should be redirected to the appropriate page that is configured in the config.

### Additional Info:
- Running Next.js 12.1.4
- All pages leverage SSG
- Page data is provided using Sanity CMS

### Example Code:

```
module.exports = {
  async redirects() {
    return [

      // This redirect from one catch all to another just 404s
      // Path: (src/pages/[...slug].tsx)
      {
        source: '/products/engage/features',
        destination: '/products/enage'
      },

      // This nested route works
      // Path: (src/pages/blog/[slug].tsx)
      {
        source: '/deleted-blog-post',
        destination: '/blog/replaced-blog-post'
      }
    ];
  }
};

```

**Comment by colevr1** [comment]
I have the same issue. Different behaviour in local and after uploading on vercel.com
Anyone has a clue for that?
